app/controllers/profiles_controller.rb

```diff
@@ -34,15 +34,10 @@ def edit
   end
 
   def update
-    update_result = ActiveRecord::Base.transaction do
-      @user.update(user_params)
-      @profile.update(profile_params)
-    end
-
     respond_to do |format|
       format.html do
-        if update_result
-          redirect_to profile_path(@user)
+        if @profile.update!(profile_params)
+          redirect_to profile_path(@user.unique_name), notice: 'Profile was successfully updated.'
```

Comment:
During the rewiring I did to make things navigate to the unique name, I defined `to_param` on the `User` model:
https://github.com/emmahyde/grouper/blob/8d4c3cb5cff26ae82de3355448d971d448a3cc98/app/models/user.rb#L46-L48

And also on the `Profile` cuz they're basically the same thing:
https://github.com/emmahyde/grouper/blob/f15e2ed3f0b0e1dcfb5d5b2a17ec48e43dd7240b/app/models/profile.rb#L21-L23

This is what all these Rails autogenerated `_path` helpers call under the hood when an ActiveRecord object is passed. The default behavior takes the id off of the object, so I've customized that behavior to pluck the `unique_name` instead.

Long story short, I think you can just do
```suggestion
          redirect_to profile_path(@user), notice: 'Profile was successfully updated.'
```

it will call `to_param` if you don't specify!

Reply:
ahhhhhhhh, that makes so much sense! I was having trouble understanding that change. Ty!